gateway-ha/src/test/java/io/trino/gateway/ha/router/TestRoutingGroupSelector.java

```diff
@@ -154,9 +143,7 @@ public void testByRoutingRulesEngineFileChange()
                         + "actions:\n"
                         + "  - \"result.put(\\\"routingGroup\\\", \\\"etl2\\\")\""); // change from etl to etl2
         fw.close();
-        long etl2LastModified = file.lastModified();
-        log.info("etl2 rule file created at: " + etl2LastModified);
-        assertNotEquals(etl2LastModified, etlLastModified);
+        file.setLastModified(lastModified + 1);
```

Comment:
Adding a number >1000 would ensure that this test works even for filesystems that only support granularity of seconds